### DISCUSSION TITLE
The future of Manim: Code or GUI

### DISCUSSION THREAD

**Discussion by Nikhil0504** [ManimCommunity/manim#1396] [General]
I think code-first should be the main thing we should work as the core of manim is the main prerequsite for making a GUI, so having as less bugs and crashes would be the first thing to work on. Moreover this would help us a lot in educating python for the students out there who are using this particular software.

**Comment by RickyC0626** [comment]
> I think code-first should be the main thing we should work as the core of manim is the main prerequsite for making a GUI, so having as less bugs and crashes would be the first thing to work on. Moreover this would help us a lot in educating python for the students out there who are using this particular software.

Not only Python, but also how to implement algorithms that will make manim perform the way developers want, in various topics and subjects. I think we should strive for manim to become the goto tool for hands-on learning, with the end result of producing beautiful visualizations/animations. Even if the users' only goal is to create something simple for class or work, they can gain other skills through the process of using manim. Maybe this comes from my bias as a visual learner, but being able to see complex math equations animated AND see how they are implemented would be a life saver for classes.

**Reply by npetrangelo** [reply]
I concur. I found Desmos to be really useful when I was in high school by the same principle; its example graphs also contain the equations that define them.

**Reply by RickyC0626** [reply]
> I concur. I found Desmos to be really useful when I was in high school by the same principle; its example graphs also contain the equations that define them.

Imagine Desmos for almost every field, all in one place.

**Comment by Hari-07** [comment]
While all the points are extremely valid, I think having a software which is, atleast as a starting point, an easy drag and drop style for simple stuff and we can edit scripts on individual components for as of yet unimplemented stuff, would be great to making manim used by more people. Editing small pieces of logic would also be less intimidating to the people new to python itself, or allow to modify the code the gui generates, it can also be a way to learn manim code by looking at the generated code

Or atleast maybe make manimce in such a way that its possible to have a parallel seperate project which builds on top of manim in this way, the real time rendering would be the major change required to do so ig.
